clippy_lints/src/write.rs

```diff
@@ -234,20 +237,30 @@ fn check_tts<'a>(cx: &EarlyContext<'a>, tts: &ThinTokenStream, is_write: bool) -
         false,
         false,
     );
+    let mut expr: Option<Expr> = None;
     if is_write {
         // skip the initial write target
-        parser.parse_expr().map_err(|mut err| err.cancel()).ok()?;
+        expr = match parser.parse_expr().map_err(|mut err| err.cancel()).ok() {
+            Some(p) => Some(p.and_then(|expr| expr)),
```

Comment:
Remove `ok()` after `map_err()`.

`and_then(|x| x)` is equivalent to [`into_inner()`](https://doc.rust-lang.org/nightly/nightly-rustc/syntax/ptr/struct.P.html#method.into_inner)

Reply:
I didn't see that in the docs :smile: .